compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -233,6 +247,8 @@ class ROCMTargetBackend final : public TargetBackend {
     std::vector<std::array<int32_t, 3>> workgroupSizes;
     SmallVector<uint32_t> workgroupLocalMemories;
     int32_t subgroupSize = 64;
+    std::string subTarget = options.targetChip.substr(0, 4);
+    const std::string GFX9("gfx9");
```

Comment:
Instead of extracting the prefix (which assumes minimum size 4), should we do `options.targetChip.strarts_with(GFX9)`? (may need to be cast to `StringRef` first if it's `std:string`)

Reply:
because this is existing code moved around, feel free to keep as-is

Reply:
sure I can change it to use starts_with instead of assuming a fixed prefix size.